File: craftbukkit/src/main/java/me/lukiiy/discordBridge/event/BridgeMinecraftReceiveEvent.java
package me.lukiiy.discordBridge.event;

import org.bukkit.entity.Player;
import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;

public class BridgeMinecraftReceiveEvent extends Event implements Cancellable {
    private boolean cancel = false;
    private final Player player;
    private String message;

    public BridgeMinecraftReceiveEvent(Player player, String message) {
        super("BridgeMinecraftReceiveEvent");
        this.player = player;
        this.message = message;
    }

    public Player getPlayer() {
        return player;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public boolean isCancelled() {
        return cancel;
    }

    public void setCancelled(boolean cancel) {
        this.cancel = cancel;
    }
}

File: paper/src/main/kotlin/me/lukiiy/discordBridge/event/BridgeMinecraftReceiveEvent.kt
package me.lukiiy.discordBridge.event

import net.kyori.adventure.text.Component
import org.bukkit.entity.Player
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class BridgeMinecraftReceiveEvent(val player: Player?, @JvmField var message: Component) : Event(), Cancellable {
    private var cancel = false

    override fun isCancelled(): Boolean = cancel

    override fun setCancelled(cancel: Boolean) {
        this.cancel = cancel
    }

    override fun getHandlers(): HandlerList = handlerList

    companion object {
        val handlerList: HandlerList = HandlerList()
    }
}